portfolio/src/main/webapp/script.js

```diff
@@ -12,26 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Fetches data from servelet when the document is loaded
-window.onload = async () => {
-  let commentResponse = undefined;
-  try {
-    commentResponse = await fetch('/comment');
-  } catch (e) {
-    alert('Cannot get response');
-    console.log(e);
-    return;
-  }
+import Controller from '/Controller.js'
 
-  try {
-    const jsonArray = await commentResponse.json();
-    const comments = jsonArray
-      .map(entry => 
-      `<div class='entry'><p> ${entry.userName} : ${entry.comment}</p></div>`)
-      .join('');
-    document.getElementById('comments').innerHTML = comments;
-  } catch (e) {
-    alert('Cannot parse response');
-    console.log(e);
-  }
+// Fetches data from servelet when the document is loaded.
+// Checks whether the user has logged in and if not, get the 
+// login url from backend.
+window.onload = async () => {
+  window.controller = new Controller(document);
+  await Controller.loadComments(document.getElementById('comments'), 
+                                document.getElementById('comment-cell'));
+  await Controller.checkLoginStatus(window.controller);
```

Comment:
Move these into the constructor of your Controller object. Or in an `init_` function or similar that is called in the Controller object. You want to keep your actual js on the webpage as minimal as possible.

Reply:
Try adding `window.controller.init()`